ts/src/alpaca.ts

```diff
@@ -298,6 +306,299 @@ export default class alpaca extends Exchange {
+         * @description fetches all the account open positions
+         * @see https://docs.alpaca.markets/reference/getallopenpositions-1
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {response} the account open positions (not-parsed into typed object)
+         */
+        const response = await this.traderPrivateGetV2Positions (params);
+        return response;
+        // [
+        //     {
+        //       "asset_id": "b0b6dd9d-8b9b-48a9-ba46-b9d54906e415",
+        //       "symbol": "AAPL",
+        //       "exchange": "NASDAQ",
+        //       "asset_class": "us_equity",
+        //       "asset_marginable": true,
+        //       "qty": "1",
+        //       "avg_entry_price": "221.88",
+        //       "side": "long",
+        //       "market_value": "226.34",
+        //       "cost_basis": "221.88",
+        //       "unrealized_pl": "4.46",
+        //       "unrealized_plpc": "0.020100955471426",
+        //       "unrealized_intraday_pl": "4.46",
+        //       "unrealized_intraday_plpc": "0.020100955471426",
+        //       "current_price": "226.34",
+        //       "lastday_price": "226.34",
+        //       "change_today": "0",
+        //       "qty_available": "1"
+        //     },
+        //     {
+        //       "asset_id": "64bbff51-59d6-4b3c-9351-13ad85e3c752",
+        //       "symbol": "BTCUSD",
+        //       "exchange": "CRYPTO",
+        //       "asset_class": "crypto",
+        //       "asset_marginable": false,
+        //       "qty": "0.767499999",
+        //       "avg_entry_price": "56579.59532038",
+        //       "side": "long",
+        //       "market_value": "44059.91849259294",
+        //       "cost_basis": "43424.839351812",
+        //       "unrealized_pl": "635.07914078094",
+        //       "unrealized_plpc": "0.0146247896425307",
+        //       "unrealized_intraday_pl": "635.07914078088532038",
+        //       "unrealized_intraday_plpc": "0.0146247896425294",
+        //       "current_price": "57407.06",
+        //       "lastday_price": "58026.611",
+        //       "change_today": "-0.0106770150681383",
+        //       "qty_available": "0.767499999"
+        //     }
+        //   ]
+    }
+
+    async fetchBalance (params = {}): Promise<Balances> {
+        /**
+         * @method
+         * @name alpaca#fetchBalance
+         * @description fetches the balance on the account, uses fetch positions
+         * @see https://docs.alpaca.markets/reference/getaccount-1
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {int} the current integer timestamp in milliseconds from the exchange server
+         */
+        // get the account level information
+        const accountInfo = await this.fetchAccountInfo ();
+        // get the positions on the account
+        const positions = await this.fetchOpenPositions ();
```

Comment:
The only way to get the balance splits by currency on the the account is to use the positions.

fetchAccountInfo returns only a total account level cash balance and a total usd denominated crypto balance.
```
"cash": "56208.81",
...
"portfolio_value": "100531.63",
```
I used both so we can grab both cash and underlying crypto denominated balances.

Reply:
FYI these are not positions in the sense of margin positions. They are coins held in the wallet.

Reply:
@alisalama in this case I think we should use the positions endpoint just if the user provides `marginMode:isolated` in `params`

Reply:
@carlosmiei I can do that, but alpaca exchange is configured for Spot only so would that be appropriate?

Perhaps fetchOpenPositions is better named fetchBalancesByCurrency or something similar? I think understandably the word "positions", taken from their api naming convention, is causing some confusion.

https://github.com/ccxt/ccxt/blob/ed5c23b522b1e4e5cbaf8a92a02a22a96a339273/ts/src/alpaca.ts#L42-L47

Reply:
Or probably even better I just remove the whole `fetchOpenPositions` and put that functionality inside the fetchBalance. Seems the easiest. No need for the separate method as it's not used elsewhere.

Reply:
@alisalama Yes I think we don't need `fetchOpenPositions` then

Reply:
But we still need to 
- by default fetch account info
- if the user provides an option, then we use openPositions instead

Reply:
> @alisalama Yes I think we don't need `fetchOpenPositions` then

Cool. Will restructure it.

> But we still need to
> 
> * by default fetch account info
> * if the user provides an option, then we use openPositions instead

I don't really follow why the 2nd part is optional. Maybe I'm being a bit dumb, but AFAIU the openPositions is required to fill the full coin denominated balances like https://github.com/ccxt/ccxt/wiki/Manual#balance-structure. Perhaps it is not required to have all the coins present in the balance?

As an example:

```
...
'free': { 
'USD' : 1000.00, // this can be retrieved from AccountInfo
'BTC' : 1.50, // can only be retrieved from openPositions (these are not positions as in margin positions, they are coins held in wallet)
'ETH' : 7.00, / can only be retrieved from openPositions (these are not positions as in margin positions, they are coins held in wallet)
},
...
```

Reply:
@alisalama the the problem is we have a rule of 1 request per unified call (with some rare exceptions)

Reply:
Ok, thanks for clarifying the issue. That was actually my original purpose for splitting the http requests out into two separate functions, perhaps that was cheating a little bit.

So the accepted solution is to add a parameter, even if the parameter will always need to be applied?

Reply:
No the goal was to perform only 1 request or the other depending on the parameter being provided or not.

But ok we can keep both requests in this case for now, but they should be done "in parallel" using `promise.all`